Sources/Testing/Attachments/Test.Attachment.swift

```diff
@@ -0,0 +1,300 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+private import _TestingInternals
+
+@_spi(Experimental)
+extension Test {
+  /// A type describing values that can be attached to the output of a test run
+  /// and inspected later by the user.
+  ///
+  /// Attachments are included in test reports in Xcode or written to disk when
+  /// tests are run at the command line. To create an attachment, you need a
+  /// value of some type that conforms to ``Test/Attachable``. Initialize an
+  /// instance of ``Test/Attachment`` with that value and, optionally, a
+  /// preferred filename to use when writing to disk.
+  public struct Attachment: Sendable {
+    /// The value of this attachment.
+    ///
+    /// The type of this property's value may not match the type of the value
+    /// originally used to create this attachment.
+    public var attachableValue: any Attachable & Sendable /* & Copyable rdar://137614425 */
```

Comment:
I know this is all `@_spi(Experimental)` right now, but _when_ we propose and eventually elevate the pieces which are intended to become API to fully `public`, could we plan to keep this property `@_spi`? Or perhaps even lower it right now from `public` to `internal` or `private`?

The fact that this is an `any` existential in the implementation doesn't bother me so much, but since it is possible we might replace that with a concrete type on some platforms in the future, it may be safer and more conservative to start out without exposing this property as proper API.

Reply:
Sure. I'm going to leave it unannotated for now to make my life a bit easier as I stage in bits of this feature in future PRs, but will keep it in mind for when we raise any of this to API.